volumes/miovision/api/pull_alert.py

```diff
@@ -133,56 +131,37 @@ def pull_alerts(conn: any, start_date: datetime, end_date: datetime, key: str):
     key : str
         Miovision API access key.
     """
-
-    STEP_SIZE = datetime.timedelta(minutes=5)
-
-    #establish list of timestamps to iterate over
+    start_date = TZ.localize(start_date)
+    end_date = TZ.localize(end_date)
     if end_date < start_date:
         raise ValueError('end_time is not greater than start_time.')
-    dt = start_date
-    dt_list = [dt, ]
-    while(dt < end_date):
-        dt = dt + STEP_SIZE
-        dt_list.append(dt)
-
     logger.info('Pulling Miovision alerts from %s to %s.', start_date, end_date)
-
-    #pull alerts at each timestamps and append to list
+    #pull alerts from each page and append to list
     dfs = []
-    for dt in dt_list: 
-        miovpull = MiovAlertPuller(dt, key)
-        test = miovpull.get_response()
-        response = miovpull.process_response(test)
-        df = pd.DataFrame(response).drop_duplicates()
+    pageSize = 100
```

Comment:
Would pagesize change in the future?

Reply:
We probably won't ever change it. The API default is 1 with maximum being 1000. I found it was much faster when set to a higher number.

Reply:
so pagesize is like a "how many alerts is shown in one page" instead of "look through 100 pages of alerts" ?

Reply:
mainly just wondering if we might miss some alerts on another page if its the latter

Reply:
ohhh no, the block below is iterating through the pages. See lines 143, 146.
The API returns the link to the next page so you just keep following the next link. 

Here's the schema of the response from the API docs: 

```
{
  "alerts": [
    {
      "alertId": "string",
      "alertStartDateTime": "2024-07-15T15:37:11.809Z",
      "alertEndDateTime": "2024-07-15T15:37:11.809Z",
      "intersectionId": "string",
      "type": "string"
    }
  ],
  "links": {
    "next": "string"
  }
}
```

Reply:
:ayeaye: